### PR TITLE
Tbolt/2638 Adds ability to select multiple states during onboarding

### LINKED ISSUE #2638: [Dev] Create ability for user to select multiple states to request access to
Currently users can only select one state at a time to request access for.

Feature -  
Change the dropdown in the state request to be an autocomplete field that allows users to select multiple options.

### This task is done when…
This could include:
 - [ ] change dropdown to an autocomplete

### PR DESCRIPTION
**Pull request by tbolt**
Resolves #2638 

### This pull request changes...

In the "Select State Affiliation" step of onboarding
- It's now using an autocomplete component instead of the previous dropdown
- Users can now select multiple state affiliations with no limit

### This is how to verify this change...
1. Checkout PR build
2. Login with an account that has not selected a state affiliation yet (norole likely)
3. Verify the new component matches design specification provided in Figma

### This pull request is ready to merge when...

- [ ] Develop has updated automated tests (and all tests are passing)
- [ ] This code has been reviewed by someone other than the original author
- [x] QA has manually tested and created bugs for any issues found (beyond minor fixes)
- [ ] The experience passes a basic manual accessibility audit (keyboard nav, screenreader, text scaling) OR an exemption is documented
- [ ] The change has been documented
- [ ] Associated OpenAPI documentation has been updated
- [x] Changelog is updated as appropriate

### This feature is done when...

- [ ] QA has verified the accessibility and functionality related to the change
- [ ] Design has approved the experience
- [ ] Product has approved the experience

### CODE REVIEW

File: web/src/containers/StateAccessRequest.js [hunk 1]
```diff
@@ -1,18 +1,67 @@
...
-import { usStatesDropdownOptions } from '../util/states';
+import { STATES } from '../util/states';
 
 const StateAccessRequest = ({ saveAction, errorMessage, fetching }) => {
-  const [selectedStates, setStates] = useState([
-    usStatesDropdownOptions[0].value
-  ]);
+  // We maintain 4 objects to support the multiple state selection
+  // 1. initialStates - full list of states that can be filtered against. as
+  // states/items are selected by the user, they are removed from this list
```
[the remark this answers is not in the document]
Comment (tbolt) [reply web/src/containers/StateAccessRequest.js:12]: ohhh, that's a great idea. I'll work on that.

**Comment by tbolt** [comment]
@thetif I updated the code to use a reducer hook. Take a look if you get a chance.

**Comment by thetif** [comment]
Just finished double checking, looks good, I like the new component.

**Comment by eStar386** [comment]
Fails Accessibility Tests verified with Wave: 
WaveFailure

**Comment by eStar386** [comment]
Per Discussion with Ty and to fix accessibility issues can the highlighted portion be changed to **Choose your State Affiliations** to pass wave check and improve accessibility.  @tbolt @thetif @beparticular @akuas @NAretakis @CSwartzHMA 
Screen Shot 2021-05-19 at 10 56 35 AM
Passes functional testing  and needs verification on requested accessibility change.

**Comment by eStar386** [comment]
Note per #3124, and documented in test case researched the aria autocomplete function to suggest changes that pass wave - https://design.cms.gov/components/autocomplete/

**Review by eStar386 — APPROVED** [review]
Passes the functional testing and documented accessibility errors/references provided in link.

**Comment by beparticular** [comment]
@tbolt from what I can tell with the Aria error and the design system, we might should be using something more like this for the autocomplete field. Do you think that's more in line with the accessibility standards for the autocomplete option? 
https://www.figma.com/file/6eVvo7JjvXiovGTR4BoVgK/CMS-eAPD-2020?node-id=2493%3A22276

Also see the frame directly above that one in Figma for another possibility.

**Review by NAretakis — APPROVED** [review]
Good to go.

**Comment by tbolt** [comment]
@eStar386 Accessibility issues should be resolved in the latest update

@beparticular I think how I have it is acceptable but maybe not ideal. I currently have the "Select State Affiliation" as just a plain <p> tag and the input field is labeled with "Select State Affiliation" but is hidden and only reads for screen readers. If I am looking at the right board-
image
This would be a better option but where would the Badges go?

**Comment by eStar386** [comment]
@tbolt Verified that passes wave. 
Adding the relevant test case: [2638](https://docs.google.com/spreadsheets/d/1H9NykPUnO-cLLozS1com2Kewe5b8YFwdIHC04o1ZGoo/edit?usp=sharing)

**Comment by beparticular** [comment]
Spoke with @tbolt and the current implementation with the fixes for accessibility work and look great and we'll stick with that, no changes to the language within the search box are needed at this time. Updating the Figma file to match.

**Review by beparticular — APPROVED** [review]

**Pull request merged**